### ISSUE TITLE
[BUG] on keyboard "Enter" always false due to type coercion

### ISSUE THREAD

**Issue by jalezi** [subhoghoshX/laureate#37]
### Description

`Input.tsx`

If user enters empty string condition is always `false`.

`Number('') === 0`
`Number.isNaN(Number('')) === false`

```javacript
// Input.tsx

  function keyDownHandler(e: any) {
    ...
    } else if (e.keyCode === 13) {
      // always false
      if (Number.isNaN(Number(dimensionBuffer))) {
        setDimensionBuffer(data + "");
      } else {
        setData(() => +dimensionBuffer);
      }
    }
  }
```

### Solution

```
function keyDownHandler(e: any) {
    ...
    } else if (e.keyCode === 13) {
      const isValidBuffer = !(!dimensionBuffer || Number.isNaN(Number(dimensionBuffer)));
      if (isValidBuffer) {
        setData(() => +dimensionBuffer);
      } else {
        setDimensionBuffer(data + "");
      }
    }
  }

**Comment by subhoghoshX** [comment]
Hey, thanks.

Actually, that's the expected behavior. To interpret empty strings as 0 (zero).

The first condition is for when a user enters some random string and presses `Enter`. I wanted the random string to be replaced with the current value of data. Hope it makes sense.

**Comment by jalezi** [comment]
Sure it makes sense for random string.

For empty string I would rather see if it resets to last value.
E.g. If user starts deleting width and suddenly realizes that he wanted to change height, the width is set to 0 (onblur if empty string) and after he enters height, he must also set width.

BTW: great job 👍

**Comment by subhoghoshX** [comment]
Yeah makes sense, just checked a popular design tool and its behavior is like you described. Feel free to work on it if you want.

> BTW: great job 👍 

Thanks, dude :)

**Comment by jalezi** [comment]
> Yeah makes sense, just checked a popular design tool and its behavior is like you described. Feel free to work on it if you want.

OK. Should I open new PR or just continue in #40?

**Comment by subhoghoshX** [comment]
> OK. Should I open new PR or just continue in #40?

It's better if you create a new one. I can see some issues with #40 i.e. too many file changes. You might need to close that as well.

**Comment by jalezi** [comment]
> > OK. Should I open new PR or just continue in #40?
> 
> It's better if you create a new one. I can see some issues with #40 i.e. too many file changes. You might need to close that as well.

Yeah, I s.... up.
